target/linux/ath79/nand/base-files/lib/upgrade/platform.sh

```diff
@@ -1,24 +1,93 @@
+#!/bin/sh
+
+# SPDX-License-Identifier: GPL-2.0
 #
-# Copyright (C) 2011 OpenWrt.org
+# Copyright (C) 2019 Jeff Kletsky
 #
 
 PART_NAME=firmware
-REQUIRE_IMAGE_METADATA=1
 
+REQUIRE_IMAGE_METADATA=1
 platform_check_image() {
 	return 0
 }
 
+RAMFS_COPY_BIN='fw_printenv fw_setenv'
+RAMFS_COPY_DATA='/etc/fw_env.config /var/lock/fw_printenv.lock'
+
```

Comment:
Hi again!

I think it doesn't worth to name this cross-flashing function so generally. It's unlikely that usual router vendors would use both a nor flash large enough for a firmware and a nand flash as well. Usually there's a 1M SPI NOR for bootloader and the entire firmware lies in SPI NAND. This function is currently only suitable for the two gl-inet routers you added here and I think you should move this function into glinet.sh and name it like "glinet_nand_nor_do_upgrade"

Reply:
I'll merge this PR myself without the two sysupgrade commits after this change.

Reply:
@981213 Consider updating the upgrade instructions in the commit message, or at least stating that those are not working then.

Reply:
All meaningful sysupgrade paths rechecked, including use of "force". SUPPORTED_DEVICES as appropriate. Matrix of test results available in comments here. Commit messages adjusted.